add progress indicator

kyon0304 opened the pull request:
添加进度条提示, 包括阅读进度和加载进度

kyon0304 committed:
add progress indicator

kyon0304 commented:
cc @hxgdzyuyi

hxgdzyuyi commented on src/views/reader/progressIndicator.js:
```diff
@@ -0,0 +1,26 @@
+
+  render() {
+    return (
+      <div>
```
有一个问题。 viewed 有可能不是断断续续的 不是连续唯一的。 比如你中途跳页了。

尽量用 class 不要用 id @kyon0304

hxgdzyuyi replied:
可以先不做 loaded

kyon0304 replied:
@hxgdzyuyi 用 class 不用 id 是出于什么样的考虑? 
中途跳页那个对 loaded 确实有影响, 不过对 viewed 还好吧.

hxgdzyuyi commented on src/models/progressIndicator.js:
```diff
@@ -0,0 +1,17 @@
+import { Model } from 'backbone'
+
+export default class extends Model {
```
percent相关的可以绑定到 book 对象上，不需要新开一个对象。 因为 progress 和 currentPage 和 totalPage 其实是一个车性质的。 cc  @kyon0304

kyon0304 replied:
@hxgdzyuyi 嗯, 好~

hxgdzyuyi commented on src/styles/reader/progressIndicator.styl:
```diff
@@ -0,0 +1,14 @@
+
+#viewed
+  background-color: #00a497
+  z-index: 101
```
可以加上 `transition:width` 之类的简单动画 cc @kyon0304

kyon0304 committed:
fix according to PR comment

1. remove loaded progress indicator
2. remove progress indicator model and global variable

kyon0304 committed:
add simple transition

kyon0304 commented:
cc @hxgdzyuyi

kyon0304 committed:
remove dead code

hxgdzyuyi commented on src/views/reader/canvas/image.js:
```diff
@@ -199,6 +198,7 @@ export default class extends React.Component {
     var book = app.getModel('book')
       , canvas = app.getModel('canvas')
       , { width, height } = book.getCurrentImage()
+      , loader = app.getModel('loader')
```
@kyon0304 这个没用到？

hxgdzyuyi commented on src/views/reader/index.js:
```diff
@@ -20,12 +21,15 @@ export default class extends React.Component {
-
   setCurrentPage(props) {
     var book = app.getModel('book')
+      , loader = app.getModel('loader')
```
@kyon0304 dead code ?

kyon0304 committed:
fix according to PR comment

1. move loader from models back to manager
2. remove dead code

kyon0304 commented:
cc @hxgdzyuyi
